components/search/BookingInfoSearch/DatePickerWithRange.tsx

```diff
@@ -37,23 +35,20 @@ type DatePickerWithRangeProps = {
     checkoutDate: string;
     numDays: number;
   }) => void;
-  defaultNumDays?: number;
+  onValidationChange?: (isValid: boolean) => void;
+  bookingInfo: searchParamsType;
 };
 
 export function DatePickerWithRange({
+  bookingInfo,
   className,
   onChange,
-  defaultNumDays = 7,
+  onValidationChange,
 }: DatePickerWithRangeProps) {
-  // Generate default check-in and check-out dates
-  const { DEFAULT_CHECKIN_BOOKING_DATE, DEFAULT_CHECKOUT_BOOKING_DATE } =
-    generateDefaultDates(defaultNumDays);
-
   const [dateRange, setDateRange] = React.useState<DateRange>({
-    from: new Date(DEFAULT_CHECKIN_BOOKING_DATE),
-    to: new Date(DEFAULT_CHECKOUT_BOOKING_DATE),
+    from: new Date(`${bookingInfo.checkinDate}T00:00:00`),
+    to: new Date(`${bookingInfo.checkoutDate}T00:00:00`),
```

Comment:
Parsing a string with `Date` will return a date in UTC.
This means that based on the time of day, the dates be off by one.

@luSteven01 Can you review this file.

Reply:
Used local time using `date-fns` parse.